documentation/library_guide/overview.rst

```diff
@@ -61,9 +61,28 @@ does (see the |dpcpp_short| specification and the SYCL specification for details
 Known Limitations
 =================
 
-For ``transform_exclusive_scan``, ``transform_inclusive_scan`` algorithms, the result of the unary operation should be
-convertible to the type of the initial value if (one is provided), otherwise it is convertible to the type of values
-in the processed data sequence: (``std::iterator_traits<IteratorType>::value_type``).
+* For ``transform_exclusive_scan``, ``transform_inclusive_scan`` algorithms, the result of the unary operation should be
+  convertible to the type of the initial value if one is provided, otherwise it is convertible to the type of values
+  in the processed data sequence: (``std::iterator_traits<IteratorType>::value_type``).
+* The definition of lambda functions used with parallel algorithms should not depend on preprocessor macros
+  that makes it different for the host and the device. Otherwise, the behavior is undefined.
+* ``exclusive_scan`` and ``transform_exclusive_scan`` algorithms may provide wrong results with
+  vector execution policies when building a program with GCC 10 and using -O0 option.
+* When used within DPC++ kernels or transferred to/from a device, a container class can only hold objects
+  whose type meets DPC++ requirements for use in kernels and for data transfer, respectively. 
+* The use of oneDPL together with the GNU C++ standard library (libstdc++) version 9 or 10 may lead to
+  compilation errors (caused by oneTBB API changes). 
+  To overcome these issues, include oneDPL header files before the standard C++ header files,
+  or disable parallel algorithms support in the standard library. 
+  For more information, please see `Intel® oneAPI Threading Building Blocks (oneTBB) Release Notes`_.
+* The ``using namespace oneapi;`` directive in a oneDPL program code may result in compilation errors
+  with some compilers including GCC 7 and earlier. Instead of this directive, explicitly use
+  ``oneapi::dpl`` namespace, or create a namespace alias. 
+* ``std::array::at`` member function cannot be used in kernels because it may throw an exception;
```

Comment:
I don't know how `std::array::at` method is different from everything else that throws. Should we instead add the section that everything that is uses with oneDPL algorithms for DPC++ execution policies should respect DPC++ limitations including exception throwing, device_copyable, recursion, etc. The wording should be more accurate.

IIRC, I already made this point during our internal discussions.

Reply:
There is already a general note somewhere in the guide, IIRC.

The std::array is "special" because we claim it is supported for use in DPC++ kernels - except for the "at" method. Maybe this note makes most sense as a remark at the "Tested API" page.

Reply:
We do the same claim for `variant` and `optional` I believe but for the methods that don't throw. I don't understand how it's different and why ``std::array`` is the "special" case.

I can see some information in the "Restrictions" section and it's good we have generic comment about that, which is basically what I suggested. I that case I would completely remove the note about `std::array` or move it to more appropriate place. However, I would add more information the the restrictions list. Exceptions to me it very basic and understandable restriction (more basic than capturing buffer into a lambda, which is already there).

I would add something like: "Calling the API that throws exception is not allowed within lambdas passed to an algorithm.

Reply:
It's not limited to lambdas. What would be a proper wider term - function objects?

Reply:
I would say Callable objects or Function Objects.

I am not sure if we have ever tested with functions and if DPC++ allows that.

Reply:
I've added the following line "Calling the API that throws exception is not allowed within callable objects passed to an algorithm."